### PR TITLE
Convert PingEquella to CheckFiles

### PR DESCRIPTION
**Pull request by cbeach47**
- Migrated https://github.com/openequella/openEQUELLA-Tools/tree/master/ping-equella to   `CheckFiles`
- Added support for oEQ advanced filestores
- 'REST' `cf.mode` is disabled for now.
- Switched DB support from SQL Server to Postgre
- Simplified the `Config` access

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # openEQUELLA Toolbox
-Proof-of-concept for a suite of scripts to interface with openEQUELLA in Java.
+A Java-based suite of scripts geared to worked with openEQUELLA.
```
Comment (bjagg) [inline README.md]: `worked` -> `work`

**Review by bjagg — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ChristianMurphy) [reply README.md]:
```suggestion
A Java-based suite of scripts geared to work with openEQUELLA.
```

File: build.gradle [hunk 2]
```diff
@@ -48,11 +47,4 @@ jar {
...
 	}
-}
-
-//distZip {
-//	into(project.name+"-"+getVersion()) {
-//		from '.'
-//		include 'conf/*'
-//	}
-//}
+}
```
Comment (ChristianMurphy) [inline build.gradle]: newline at end of file

```suggestion
}

```

File: conf/blank.properties [hunk 3]
```diff
@@ -61,4 +61,88 @@ jsonreport.sources=
...
+# uuid 123456-7890-1234-567890, and adv filestore path /my/extra/data,  this line would look like:
+# cf.filestore.adv.filestore.23.123456-7890-1234-567890=/my/extra/data/Institutions/CfTesting/Attachments/123456-7890-1234-567890
+# Note:  Since this is a 'dynamic' property, it won't show up in the property list at runtime.
+cf.filestore.advanced.{inst-id}.{collection-uuid}=
+
+# Will compare the previous error report with the error report just ran.
+# Recommended to use in conjuction with email for an automated regular cron job
+# Values: true|false
+# Default:  false
+cf.compare.missing.atts.after.run=
```
Comment (ChristianMurphy) [inline conf/blank.properties]: newline at end of file

```suggestion
cf.compare.missing.atts.after.run=

```

File: conf/junit.properties [hunk 4]
```diff
@@ -0,0 +1,10 @@
+email.username=
+email.password=
+cf.email.recipients=
+cf.db.url=
+cf.db.username=
+cf.db.password=
+cf.filestore.directory=
+cf.filestore.advanced.812.44ac1c1a-dd0c-462f-b717-53324c0dd6f9=
+cf.filestore.advanced.1509.44ac1c1a-dd0c-462f-b717-53324c0dd6f9=
+cf.filestore.advanced.812.5a7b2083-2671-414b-8e5b-c4cd9dfa1f30=
```
Comment (ChristianMurphy) [inline conf/junit.properties]: newline at end of file

```suggestion
cf.filestore.advanced.812.5a7b2083-2671-414b-8e5b-c4cd9dfa1f30=

```

File: src/main/java/org/apereo/openequella/tools/toolbox/utils/CheckFilesUtils.java [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+	/**
+	 * Lines parsed into: Item UUID Item Version ItemStatus Collection UUID
+	 * Attachment Type Attachment UUID Attachment Status Attachment Response
+	 * Code Item Name *** Attachment Filepath ***
+	 *
+	 * *** Quoted CSV data. not used in comparisons, thus not set.
+	 *
+	 * @param row
+	 * @returnt
+	public void testFindOnlyInFirst() {
```
Comment (ChristianMurphy) [inline src/main/java/org/apereo/openequella/tools/toolbox/utils/CheckFilesUtils.java]: Is this function intended to be commented out?

File: src/test/java/org/apereo/openequella/tools/toolbox/CheckFilesRestTests.java [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+//			assertTrue(PingEquellaDriver.run());
+//			assertTrue(PingEquellaDriver.finalizeRun());
+//
+//			//ALL results
+//			File allResults = new File(ReportManager.getInstance().getStdOutFilename());
+//			String filename = Config.getInstance().getTestDataDir()+"/outputSamplesToCompare/ping_attachments_junitTestsAttachmentRun1_all_stats_GOLD.csv";
+//			File allResultsToCompare = new File(filename);
+//			assertTrue(allResults.exists());
+//			assertTrue(allResultsToCompare.exists());
+//			assertTrue(PingUtils.mediumStrictCompare(allResults, allResultsToCompare));
```
Comment (ChristianMurphy) [inline src/test/java/org/apereo/openequella/tools/toolbox/CheckFilesRestTests.java:51]: Are these tests intended to be commented out?

File: src/test/java/org/apereo/openequella/tools/toolbox/TestUtils.java [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+//	public static void resetTestOutputDirectory() {
+//		File dir = new File(Config.get(Config.CF_OUTPUT_FOLDER));
+//		String[] errorStats = dir.list(new FilenameFilter() {
+//			@Override
+//			public boolean accept(File file, String filename) {
+//				return !filename.equals("placeholder");
+//			}
+//		});
+//
+//		for(String f : errorStats) {
```
Comment (ChristianMurphy) [inline src/test/java/org/apereo/openequella/tools/toolbox/TestUtils.java]: Is this intended to be commented out?

File: src/test/resources/checkFiles/misc/Institutions [hunk 8]
```diff
@@ -0,0 +1 @@
+naughty naughty
```
Comment (ChristianMurphy) [inline src/test/resources/checkFiles/misc/Institutions]:
```suggestion
naughty naughty

```

**Comment by cbeach47** [comment]
Also, pork-barreled a new simple function - AttachmentHash

File: src/test/java/org/apereo/openequella/tools/toolbox/TestUtils.java (hunk 7 above)
Comment (cbeach47) [reply src/test/java/org/apereo/openequella/tools/toolbox/TestUtils.java]: thanks for the catch.  Should have been removed.

File: src/test/java/org/apereo/openequella/tools/toolbox/CheckFilesRestTests.java (hunk 6 above)
Comment (cbeach47) [reply src/test/java/org/apereo/openequella/tools/toolbox/CheckFilesRestTests.java:51]: yes, that's correct.  The `REST` CheckFiles mode was not converted to oeq-toolbox yet.

File: src/main/java/org/apereo/openequella/tools/toolbox/utils/CheckFilesUtils.java (hunk 5 above)
Comment (cbeach47) [reply src/main/java/org/apereo/openequella/tools/toolbox/utils/CheckFilesUtils.java]: thanks for the catch, I'll remove it.

**Pull request merged**